lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp

```diff
@@ -213,6 +215,158 @@ ros::Duration LCIStrategicPlugin::get_earliest_entry_time(double remaining_dista
 
 }
 
+ros::Time LCIStrategicPlugin::get_final_entry_time_and_conditions(const VehicleState& current_state, const ros::Time& earliest_entry_time, lanelet::CarmaTrafficSignalPtr traffic_light, bool& is_entry_time_within_green_or_tdb, bool& in_tdb)
```

Comment:
I'm assuming this code is essentially entirely code moved from lci_strategic_plugin.cpp's main body? That this was just lifted from the maneuver planning callback to here so that it can be referenced? Or are there more substantial changes in the body of this code?

Reply:
Yep! This is just refactor and doesn't have any changes in the code.